### PR TITLE
fix: [CURLRequest] Multiple HTTP 100 return by API.

### LINKED ISSUE #8347: Bug: [CURLRequest] Multiple HTTP 100 return by API
### PHP Version

8.1

### CodeIgniter4 Version

4.4.2

### CodeIgniter4 Installation Method

Composer (using `codeigniter4/appstarter`)

### Which operating systems have you tested for this bug?

Windows, Linux

### Which server did you use?

apache

### Database

_No response_

### What happened?

When I call API built with a Flask server, this one return these headers:

HTTP/1.1 100 Continue

HTTP/1.1 100 Continue

HTTP/1.1 500 INTERNAL SERVER ERROR
Server: Werkzeug/2.2.3 Python/3.7.16
Date: Tue, 19 Dec 2023 21:30:20 GMT
Content-Type: application/json
Content-Length: 403
Access-Control-Allow-Origin: *
Connection: close

In this case the CURLRequest class doesn't parse correctlly the headers

### Steps to Reproduce

Make POST API with Flask 2.2.5 in debug mode and call it via CURLRequest

### Expected Output

In this case status code should be 500 not 100

### Anything else?

Just change this:
```php
        if (strpos($output, 'HTTP/1.1 100 Continue') === 0) {
```
By this:
```php
        while (strpos($output, 'HTTP/1.1 100 Continue' . $breakString) === 0) {
```

at line 392

For others who have the problem, actually I overwrite the class and specially the function send in my project and instanciate it via this function in Config\Service:

```php
    public static function curlrequest($options = [], ResponseInterface $response = null, App $config = null, $getShared = false) {
        if ($getShared) {
            return static::getSharedInstance('curlrequest');
        }
        
        if(empty($config)) {
            $config = config(App::class);
        }
        return new CURLRequest($config, new URI((empty($options['baseURI']) ? null : $options['baseURI'])), $response, $options);
     }
```

### PR DESCRIPTION
**Pull request by ping-yee**
**Description**
Closes #8347 

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by michalsn** [comment]
Since you were able to reproduce the issue, can you also provide a test case for this scenario? Thanks!

**Comment by ping-yee** [comment]
I'm afraid I can't do that, as this seems like a specific error that occurs when using a Python framework.

To write a test case for this scenario, we would need to set up a Flask API service in advance, which I don't think is a good direction.

However, I can provide the code I used when reproducing the issue.

**Reproduce**

In CI4

```php
<?php

namespace App\Controllers;

use CodeIgniter\API\ResponseTrait;

class Home extends BaseController
{
    use ResponseTrait;

    public function index()
    {
        $client = \Config\Services::curlrequest();

        try {
            $options = [
                'headers' => [
                    'Expect' => '100-continue'
                ],
                'debug'   => true,
            ];

            $response = $client->request('GET', 'http://localhost:5000', $options);
            // We'll get 100 until we fix it.
            // After modification we will get 200.
            dd($response->getStatusCode());
        } catch (\Exception $e) {
            echo "Request failed: " . $e->getMessage();
        }
    }
}

```

In flask

```python
from flask import Flask, jsonify

app = Flask(__name__)

@app.route('/')
def hello_world():
    return jsonify(message="Hello, World!")

if __name__ == '__main__':
    app.run(debug=True, host='0.0.0.0', port=5000)
```

Also, you can get the repeat response by using curl command.

```
curl localhost:5000 -v -H "Expect: 100-continue"
```

```log
user@Py:/mnt/d/CodeIgniter4$ curl localhost:5000 -v -H "Expect: 100-continue" 
*   Trying 127.0.0.1:5000...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 5000 (#0)
> GET / HTTP/1.1
> Host: localhost:5000
> User-Agent: curl/7.68.0
> Accept: */*
> Expect: 100-continue
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 100 Continue
* Mark bundle as not supporting multiuse
< HTTP/1.1 100 Continue
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Server: Werkzeug/2.2.2 Python/3.7.17
< Date: Sun, 28 Jan 2024 06:05:36 GMT
< Content-Type: application/json
< Content-Length: 33
< Connection: close
<
{
  "message": "Hello, World!"
}
* Closing connection 0
```

**Comment by michalsn** [comment]
We don't need Flask. We tested output before, like here: https://github.com/codeigniter4/CodeIgniter4/blob/develop/tests/system/HTTP/CURLRequestTest.php#L385

**Comment by kenjis** [comment]
We use `MockCURLRequest` for testing:
https://github.com/codeigniter4/CodeIgniter4/blob/d72a893a7519a2022138ce69e698bee479d86c02/tests/system/HTTP/CURLRequestTest.php#L49

**Comment by ping-yee** [comment]
@michalsn @kenjis thank you for your explanation!
I have added the test case, review please.

File: tests/system/HTTP/CURLRequestTest.php [hunk 1]
```diff
@@ -1109,4 +1109,24 @@ public function testUserAgentOption(): void
...
+        $output = 'HTTP/1.1 100 Continue
+        Mark bundle as not supporting multiuse
+        HTTP/1.1 100 Continue
+        Mark bundle as not supporting multiuse
+        HTTP/1.1 200 OK
+        Server: Werkzeug/2.2.2 Python/3.7.17
+        Date: Sun, 28 Jan 2024 06:05:36 GMT
+        Content-Type: application/json
+        Content-Length: 33
+        Connection: close';
```
Comment (kenjis) [inline tests/system/HTTP/CURLRequestTest.php:1124]: The out is the following. This is not a valid HTTP response.
```
HTTP/1.1 100 Continue
        Mark bundle as not supporting multiuse
        HTTP/1.1 100 Continue
        Mark bundle as not supporting multiuse
        HTTP/1.1 200 OK
        Server: Werkzeug/2.2.2 Python/3.7.17
        Date: Sun, 28 Jan 2024 06:05:36 GMT
        Content-Type: application/json
        Content-Length: 33
```
And it is better to add comment showing the line break code is `CRLF`.

Comment (ping-yee) [reply tests/system/HTTP/CURLRequestTest.php:1124]: Ok I'll handle this at this weekend.

File: tests/system/HTTP/CURLRequestTest.php [hunk 2]
```diff
@@ -1109,4 +1109,30 @@ public function testUserAgentOption(): void
...
+Mark bundle as not supporting multiuse
+HTTP/1.1 200 OK
+Server: Werkzeug/2.2.2 Python/3.7.17
+Date: Sun, 28 Jan 2024 06:05:36 GMT
+Content-Type: application/json
+Content-Length: 33\r\n\r\n" . $jsonBody;
+
+        $this->request->setOutput($output);
+
+        $response = $this->request->request('get', 'http://example.com');
```
Comment (kenjis) [inline tests/system/HTTP/CURLRequestTest.php:1132]:
```suggestion
        $response = $this->request->request('GET', 'http://example.com');
```

**Review by kenjis — APPROVED** [review]

**Pull request merged**